great_expectations/data_context/store/checkpoint_store.py

```diff
@@ -63,6 +63,14 @@ def _convert_raw_json_to_object_dict(data: dict) -> dict:
     def serialize(self, value):
         # In order to enable the custom json_encoders in Checkpoint, we need to set `models_as_dict` off  # noqa: E501
         # Ref: https://docs.pydantic.dev/1.10/usage/exporting_models/#serialising-self-reference-or-other-models
+        diagnostics = value.is_added()
+        if self.cloud_mode:
+            # Cloud provides the parent ID
+            diagnostics.raise_for_errors_except_parent()
+        else:
+            # File/ephemeral will add the ID prior to this step
+            diagnostics.raise_for_errors()
```

Comment:
Ugh I don't love this but I rather this than override the public JSON method?

Reply:
Yeah, I think I like this approach more